src/charm.py

```diff
@@ -346,39 +379,43 @@ def _get_public_uri(self) -> str | None:
         Returns:
             Penpot public URI.
         """
-        return self.ingress.url
+        return (
+            None if self.ingress.url is None else self.ingress.url.replace("http://", "https://")
```

Comment:
Shouldn't the ingress return the URL with HTTPS when it's configured with TLS ?
If that's not the case, probably more a bug on the ingress provider then

Reply:
Penpot, especially with OpenID Connect, doesn't work without HTTPS. And yes, ingress will return an HTTPS URL when HTTPS is enabled.

However, sometimes, for instance, if the Kubernetes ingress is behind another external ingress, like Cloudflare or the `content-cache` charm, the HTTPS setting is not applied at the Kubernetes ingress level, but rather at Cloudflare or within the `content-cache` charm. At that point, the URL returned by the ingress will not be accurate.

On another point, if a user really did not set up any HTTPS, they will face many non-specific errors in penpot. For a user with no experience with penpot, this can definitely cause confusion.

What do you think we should do to handle this situation?

Reply:
We decided with Weii to block the charm if the ingress URL is not HTTPS. This will influence the way and possibilities of deployment, but the charm shouldn't adapt to how we want to deploy. It's the deployment that needs to match the charm requirements.

Reply:
Gotcha, I have updated the charm to include a new check that requires a HTTPS ingress.